### DISCUSSION TITLE
erreurs récurrences reverso

### DISCUSSION THREAD

**Discussion by AKGfra** [MartinPELCAT/ProjetVoltaireCheat#2] [General]
Bonjour

J'ai récemment vu que vous avez mis à jour l'extension. J'ai souhaité voir si le download l'était aussi. Il fonctionne donc là n'est pas le problème.
J'aurai aimé savoir si ton extension utilise la version online ou offline de reverso, car celui de ton extension ne détecte pas certaines règles que la version en ligne détecte, comme ci joint : 
![image](https://github.com/MartinPELCAT/ProjetVoltaireCheat/assets/102437575/49944137-2ba9-4d4c-8442-b1a3153b3f16)
![image](https://github.com/MartinPELCAT/ProjetVoltaireCheat/assets/102437575/9d8dbd99-de9b-49b0-8fa4-d38dc2988bf5)

Merci pour ton implication et pour cet outil purement pédagogique

**Comment by AKGfra** [comment]
Pareil pour voire / voir

Je vous laisse regarder

**Comment by MartinPELCAT** [comment]
@AKGfra Hello,

ça utilise la version online de reverso. Tu peux le voir dans le dossier src -> popup -> src -> App.tsx
```typescript
const fetchSentence = useCallback(async (string: string) => {
    return axios.post("https://orthographe.reverso.net/api/v1/Spelling", {
      language: "fra",
      text: string,
      autoReplace: true,
      interfaceLanguage: "fr",
      locale: "Indifferent",
      origin: "interactive",
      generateSynonyms: false,
      getCorrectionDetails: true,
    });
  }, []);
``` 

Le truc c'est que l'extension fait un check sur la reponse de l'api 


```shell
curl 'https://orthographe.reverso.net/api/v1/Spelling/' \
  -H 'authority: orthographe.reverso.net' \
  -H 'accept: text/json' \
  -H 'accept-language: en-US,en;q=0.5' \
  -H 'content-type: application/*+json' \
  -H 'origin: https://www.reverso.net' \
  -H 'referer: https://www.reverso.net/' \
  -H 'sec-ch-ua: "Not A(Brand";v="99", "Brave";v="121", "Chromium";v="121"' \
  -H 'sec-ch-ua-mobile: ?0' \
  -H 'sec-ch-ua-platform: "macOS"' \
  -H 'sec-fetch-dest: empty' \
  -H 'sec-fetch-mode: cors' \
  -H 'sec-fetch-site: same-site' \
  -H 'sec-gpc: 1' \
  -H 'user-agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36' \
  --data-raw '{"englishDialect":"indifferent","autoReplace":true,"getCorrectionDetails":true,"interfaceLanguage":"fr","locale":"","language":"fra","text":"Les joueurs peuvent faire leurs pronostiques en ligne.","originalText":"","spellingFeedbackOptions":{"insertFeedback":true,"userLoggedOn":false},"origin":"interactive","isHtml":false,"IsUserPremium":false}' \
  --compressed
``` 

Et dans la réponse, il ne marque pas d'erreur.
Malheureusement je ne peux pas assurer que l'extension marche à tous les coups.
Sauf si on source sur une autre api qui à un meilleur correcteur

Par exemple de mon coté je n'ai pas l'avertissement d'erreur
![image](https://github.com/MartinPELCAT/ProjetVoltaireCheat/assets/35460122/7ec60efb-142a-4030-ba8c-f337ccb78ebd)

**Reply by AKGfra** [reply]
Bonjour

As-tu essayé de trouver une version plus récente de l'api de reverso? (ici la v1 est vraiment éclaté, il y a quelques personnes, comme s0ftik qui se sont amusé à taper les dernières versions de l'api de reverso : https://www.npmjs.com/package/reverso-api  sinon il y a aussi le site language tool : https://languagetool.org/http-api/#/default (voici leur api, il y a une version payante et une partie gratuite, à voir pour bidouiller.

Perso, je ne suis pas assez bon pour voir si ça marche, ce n'est pas ma formation donc je te laisse regarder, c'est toi l'expert de nous deux :). (PS un dernier truc pour l'interface, ça serait cool si on pouvait redimensionner la box, sur mon extension elle prend bien 30% en largeur et 20% en hauteur)

Bonne journée

**Reply by AKGfra** [reply]
https://languagetool.org/fr/dev

**Comment by farescrbs** [comment]
Salut, je pense que chatgpt ne fait pas ce genre de faute, même sur la version gratuite 3.5.
Tu serais mettre l'api de chatgpt a la place de de reverso parce que en effet c'est de la merde reverso.
Si tu y arrives je pense c'est game changer
